### PR TITLE
Adds workflow to automatically build github-pages site

### PR DESCRIPTION
**Pull request by s9latimm**
- [x] Implementation
  - [x] Python
  - [x] HTML
  - [x] JavaScript / CSS
- [x] Workflow
  - [x] Deploy
  - [x] Only `push: branches: [ master ]`
  - [x] Remove `retention-days`

### CODE REVIEW

File: .github/workflows/pages.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
```
Comment (ChristianKaltenecker) [inline .github/workflows/pages.yml]: Is there a specific reason why we use Ubuntu 20.04? Couldn't we also use Ubuntu 22.04?

**Review by ChristianKaltenecker — APPROVED** [review]
I have just a minor question regarding the Ubuntu Version in the workflow.

The rest looks good to me.
Nice job!

File: .github/workflows/pages.yml (hunk 1 above)
Comment (vulder) [reply .github/workflows/pages.yml]: @s9latimm If it works, we could also move to 22.04

File: .github/pages/index.html [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        <script type="text/javascript">
+            function render_index(index) {
+                Object.values(index).forEach(feature_model => {
+                    const tr = document.createElement('tr');
+                    tr.className = 'item default';
+                    {
+                        const td = document.createElement('td');
+                        td.innerHTML = `
+                            <div class="data">
+                                ${feature_model['name']}
```
Comment (vulder) [inline .github/pages/index.html:89]: Maybe we should make the name a bit larger? css item?

**Review by vulder — APPROVED** [review]

File: .github/pages/index.html [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                <col style="width:30%">
+                <col style="width:5%">
+            </colgroup>
+            <thead>
+            <tr>
+                <th>Name</th>
+                <th>Metadata</th>
+                <th>Files</th>
+                <th>Preview</th>
+                <th>Links</th>
```
Comment (vulder) [inline .github/pages/index.html:74]:
```suggestion
                <th>Edit</th>
```
Maybe we should call this edit, as it opens the gh editor?

Comment (s9latimm) [reply .github/pages/index.html:74]: This column may provide, for example, links to the original project source and so on.

Comment (vulder) [reply .github/pages/index.html:74]: ok, then let's keep the old wording

**Review by ChristianKaltenecker — APPROVED** [review]
Thanks for addressing my question. :+1:

File: .github/workflows/pages.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
+name: Configurable Systems Pages
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (vulder) [inline .github/workflows/pages.yml]:
```suggestion
```

**Review by vulder — APPROVED** [review]

**Pull request merged**